R/derive_var_atoxgr.R

```diff
@@ -0,0 +1,287 @@
+
+    # append lab data just graded to output data
+    out_data <- bind_rows(out_data, grade_this_term)
+  }
+
+  all_data <- out_data
+}
+
+
+#' Derive Lab High toxicity grade 0 - 4 and Low toxicity grades 0 - (-4)
+#'
+#' @description
+#'
+#' Derives character lab grade based on high and low severity/toxicity grade(s).
+#'
+#' @param dataset Input data set
+#'
+#'   The columns `ATOXGRL`, `ATOXGRH` and specified by `lotox_description_var`,
+#'   and `hitox_description_var` parameters are expected.
+#'
+#' @param lotox_description_var Variable containing the toxicity grade description
+#' for low values, eg. "Anemia"
+#'
+#' @param hitox_description_var Variable containing the toxicity grade description
+#' for low values, eg. "Hemoglobin Increased".
+#'
+#' @details
+#' Created variable `ATOXGR` will contain values "-4", "-3", "-2", "-1" for low values
+#' and "1", "2", "3", "4" for high values, and will contain "0" if value is gradable
+#' and does not satisfy any of the criteria for high or low values. ATOXGR is set to
+#' missing if information not available to give a grade.
+#'
+#' @author Gordon Miller
+#'
+#' @return The input data set with the character grade added
+#'
+#' @keywords der_bds_findings
+#'
+#' @family der_bds_findings
+#'
+#' @export
+#'
+#' @examples
+#' library(dplyr, warn.conflicts = FALSE)
+#'
+#' adlb <- tibble::tribble(
+#'   ~ATOXDSCL,          ~ATOXDSCH,        ~ATOXGRL,      ~ATOXGRH,
+#'   "Hypoglycemia",     "Hyperglycemia",  NA_character_, "0",
+#'   "Hypoglycemia",     "Hyperglycemia",  "0",           "1",
+#'   "Hypoglycemia",     "Hyperglycemia",  "0",           "0",
+#'   NA_character_,      "INR Increased",  NA_character_, "0",
+#'   "Hypophosphatemia", NA_character_,    "1",           NA_character_,
+#' )
+#'
+#' derive_var_atoxgr(adlb)
+derive_var_atoxgr <- function(dataset,
+                              lotox_description_var = ATOXDSCL,
+                              hitox_description_var = ATOXDSCH) {
+  lotox_description_var <- assert_symbol(enquo(lotox_description_var))
+  hitox_description_var <- assert_symbol(enquo(hitox_description_var))
+
+  assert_data_frame(
+    dataset,
+    required_vars = vars(
+      !!lotox_description_var,
+      ATOXGRL,
+      !!hitox_description_var,
+      ATOXGRH
+    )
+  )
+
+  lowgrade_char <- unique(dataset$ATOXGRL)
+  assert_character_vector(lowgrade_char, values = c("0", "1", "2", "3", "4", NA_character_))
+
+  highgrade_char <- unique(dataset$ATOXGRH)
+  assert_character_vector(highgrade_char, values = c("0", "1", "2", "3", "4", NA_character_))
+
+
+  # High and low missing - overall missing
+  # Low grade not missing and > 0 - overall holds low grade
+  # High grade not missing and > 0 - overall holds high grade
+  # (Only high direction OR low direction is NORMAL) and high grade normal - overall NORMAL
+  # (Only low direction OR high direction is NORMAL) and low grade normal - overall NORMAL
+  # otherwise set to missing
```

Comment:
I think these derivation rules will be useful for the users too, can we add them to the details section?

Reply:
Good idea, done.